### PR TITLE
Arrays r_dist and r_ray_num in obspy.taup.seismic_phase.calc_time() are too small

### PR DESCRIPTION
**Pull request by krischer**
Running get_ray_paths crashes for certain epicentral distances (error message below) in a homegrown TauP model. The problem is related to the hard-coded size of arrays r_dist and r_ray_num in obspy.taup.seismic_phase.calc_time(), which is fixed to 10 arrivals per phase:

 https://github.com/obspy/obspy/blob/master/obspy/taup/seismic_phase.py#L1009-L1015

Setting the size to something larger than e.g. 20 resolves the problem.

Cheers,

Bernhard 

Error message:
```
ar = mt.get_ray_paths(172.8000,46.762440693494824,["SS"])

/import/soin-data/bernhard/Software/obspy/obspy/taup/tau.pyc in get_ray_paths(self, source_depth_in_km, distance_in_degree, phase_list, receiver_depth_in_km)
    481         rp = TauPPath(self.model, phase_list, source_depth_in_km,
    482                       distance_in_degree, receiver_depth_in_km)
--> 483         rp.run()
    484         return Arrivals(sorted(rp.arrivals, key=lambda x: x.time),
    485                         model=self.model)

/import/soin-data/bernhard/Software/obspy/obspy/taup/taup_time.pyc in run(self)
     37         """
     38         self.depth_correct(self.source_depth, self.receiver_depth)
---> 39         self.calculate(self.degrees)
     40 
     41     def depth_correct(self, depth, receiver_depth=None):

/import/soin-data/bernhard/Software/obspy/obspy/taup/taup_path.pyc in calculate(self, degrees)
     22         self.recalc_phases()
     23         self.arrivals = []
---> 24         self.calculate_path(degrees)
     25 
     26     def calculate_path(self, degrees):

/import/soin-data/bernhard/Software/obspy/obspy/taup/taup_path.pyc in calculate_path(self, degrees)
     32         self.degrees = degrees
     33         for phase in self.phases:
---> 34             self.arrivals += phase.calc_path(degrees)

/import/soin-data/bernhard/Software/obspy/obspy/taup/seismic_phase.pyc in calc_path(self, degrees)
   1159         Only calls :meth:`calc_path_from_arrival`.
   1160         """
-> 1161         arrivals = self.calc_time(degrees)
   1162         for arrival in arrivals:
   1163             self.calc_path_from_arrival(arrival)

/import/soin-data/bernhard/Software/obspy/obspy/taup/seismic_phase.pyc in calc_time(self, degrees)
   1030         for _i in range(phase_count):
   1031             arrivals.append(self.refine_arrival(
-> 1032                 degrees, r_ray_num[_i], r_dist[_i], REFINE_DIST_RADIAN_TOL,
   1033                 self._settings["max_recursion"]))
   1034         return arrivals

IndexError: index 10 is out of bounds for axis 0 with size 10
```

### CODE REVIEW

**Comment by krischer** [comment]
This issue is now a PR fixing it together with a simple regression test.

File: obspy/taup/tests/test_seismic_phase.py [hunk 1]
```diff
@@ -59,6 +69,20 @@ def test_shoot_middle_ray_param(self):
             self.assertAlmostEqual(phase.time[i + 1], max_rp_arrival.time,
                                    delta=time_tol)
 
+    def test_many_identically_named_phases(self):
+        """
+        Regression test to make sure obspy.taup works with models that
+        produce many identially names seismic phases.
+        """
+        folder = tempfile.mkdtemp()
```
Comment (QuLogic) [inline obspy/taup/tests/test_seismic_phase.py]: Shouldn't we use the `TemporaryDirectory` context so it gets cleanup up?

File: obspy/taup/tests/test_seismic_phase.py [hunk 2]
```diff
@@ -59,6 +69,20 @@ def test_shoot_middle_ray_param(self):
             self.assertAlmostEqual(phase.time[i + 1], max_rp_arrival.time,
                                    delta=time_tol)
 
+    def test_many_identically_named_phases(self):
+        """
+        Regression test to make sure obspy.taup works with models that
+        produce many identially names seismic phases.
```
Comment (QuLogic) [inline obspy/taup/tests/test_seismic_phase.py]: identically named

File: CHANGELOG.txt [hunk 3]
```diff
@@ -40,6 +40,9 @@
  - obspy.signal:
    * PPSD: fix warning message on Python 3 that gets shown when waveforms and
      metadata mismatch (see #1506)
+ - obspy.taup:
+   * Fix an issue when certain models produce more than 10 phases with
```
Comment (QuLogic) [inline CHANGELOG.txt]: I'm not sure I'd call it "fixing" as opposed to "raising the limits".

**Comment by megies** [comment]
Adjusted to comments, rebased + force-pushed.

**Comment by krischer** [comment]
Thanks @megies!

**Comment by krischer** [comment]
And of course thanks @QuLogic for the review :-)

**Pull request merged**